plugins/experimental/ssl_session_reuse/src/Config.h

```diff
@@ -0,0 +1,106 @@
+/** @file
+
+  Config.h - configuration file support
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+ */
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <sstream>
+#include <mutex>
+
+struct fromstring {
+  fromstring(const std::string &string) : _string(string) {}
+
+  template <typename _T> operator _T() const
+  {
+    _T t;
+    std::istringstream ss(_string);
+    ss >> t;
+    return t;
+  }
+
+protected:
+  std::string _string;
+};
+
+class Config
+{
+public:
+  virtual bool loadConfig(const std::string &filename);
+
+  static Config &
+  getSingleton()
+  {
+    static Config onlyInstance;
+    return onlyInstance;
+  }
```

Comment:
It crossed my mind that a static singleton might not work in combination with dynamic linking, but testing indicates it's not a problem.  Seems to works if the singleton is static local var like here, or a private class static variable.

https://github.com/ywkaras/MiscRepo/tree/master/DLL_SINGLETON

```
wkaras ~/REPOS/OS_MISC/DLL_SINGLETON
C$ INLINE_FUNC_SINGLETON=0 . run.sh
static memeber singleton
match
wkaras ~/REPOS/OS_MISC/DLL_SINGLETON
C$ INLINE_FUNC_SINGLETON=1 . run.sh
inline function singleton
match
wkaras ~/REPOS/OS_MISC/DLL_SINGLETON
C$ 
```

Reply:
For what reason?